### PR TITLE
Some message printing improvements, flake8 warnings

### PR DESCRIPTION
**Pull request by whot**
No real functional changes here, just a bit of polish for the output and less flake8 complaints.

### CODE REVIEW

File: mkosi/__init__.py
Comment (behrmann) [inline mkosi/__init__.py:610]: Not that this is at the moment an issue, but this is the difference between black and flake8 and at some point I would like to see black as consistent formatting in this codebase and I'd rather switch this warning off than accomodate flake8 here.

File: mkosi/__init__.py
Comment (behrmann) [inline mkosi/__init__.py:155]: This is an an excellent change, but I think it could go a bit further. I think it would make sense to subsume all calls to `sys.stderr.write` (mostly in the summary) in a `MkosiPrinter` class and maybe also get rid of the calls to `print()` here and there, so that everything goes through the same machinery.

**Review by behrmann — COMMENTED** [review]
Thank you for the contribution!

**Comment by DaanDeMeyer** [comment]
I'd prefer if we enforced this in CI but most of these changes look good so let's merge.

@behrmann I'd merge a PR reformatting all the code with black and adding it to the CI so we don't regress. If we do that, we should also add a file that can be passed to the --ignore-revs-file git blame option containing that commit so we can still have an effective git blame.

File: mkosi/__init__.py
Comment (DaanDeMeyer) [reply mkosi/__init__.py:155]: I tend to agree here, if we do this, let's do it properly.

**Comment by behrmann** [comment]
> @behrmann I'd merge a PR reformatting all the code with black and adding it to the CI so we don't regress. If we do that, we should also add a file that can be passed to the --ignore-revs-file git blame option containing that commit so we can still have an effective git blame.

I'll get to that tomorrow.

File: mkosi/__init__.py
Comment (whot) [reply mkosi/__init__.py:155]: any thoughts on going through the python logging facilities? That'd be an easy option to control formatting and have a distinct stream of outputs.

Comment (behrmann) [reply mkosi/__init__.py:155]: Definitely open to it. I think I still have a branch lying around where I had a go at it, but I was unhappy with the log levels and messing around with handlers to remove the terminal control sequences when not needed. I think moving everything to a single printer would be a sensible step to move to logging later.

**Comment by whot** [comment]
I've updated this now with an `MkosiPrinter` class, replacing all the `sys.stderr.write` calls in the file. This is almost ready for replacing with python logging in the future, the structure is quite similar after all.

A heads-up though: I'll be afk until the new year from tonight onwards, so feel free to ignore this until then if it's not merge-ready.

File: mkosi/__init__.py
Comment (whot) [reply mkosi/__init__.py]: fixed in the latest revision with the `MkosiPrinter` class, we can swap that out for a python logger in the future.

File: mkosi/__init__.py
Comment (DaanDeMeyer) [inline mkosi/__init__.py:151]: If we're doing refactoring anyway, let's take advantage of all the work @behrmann did to allow splitting mkosi up into multiple files and move MkosiPrinter into its own file.

File: mkosi/__init__.py
Comment (DaanDeMeyer) [inline mkosi/__init__.py:162]: Saving letters doesn't matter here, better to call it _print

**Comment by DaanDeMeyer** [comment]
Overall this looks pretty good!

File: mkosi/__init__.py
Comment (whot) [reply mkosi/__init__.py]: done

File: mkosi/__init__.py
Comment (whot) [reply mkosi/__init__.py]: moved to `printer.py`

**Review by DaanDeMeyer — APPROVED** [review]
LGTM but let's wait for @behrmann's approval as well.

**Review by behrmann — APPROVED** [review]
LGTM

**Pull request merged**
